cfg/validate.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cfg
+
+import (
+	"fmt"
+)
+
+func isValidLogRotateConfig(config *LogRotateLoggingConfig) error {
```

Comment:
I see different ways of writing validation in yaml_parser. Let's use consistent format like this:

https://github.com/GoogleCloudPlatform/gcsfuse/blob/9d292fe23d1362deae2c9bfa9364f56dafcb73ed/internal/config/yaml_parser.go#L75